File: kinotic-js/workspace/packages/management-api/src/api/model/OrganizationStorage.ts
import type { DeploymentStatus } from '@/api/model/DeploymentStatus'

/**
 * An organization's storage: the one account everything its deployments publish goes to,
 * and where that account is in its lifecycle. The account's details fill in as provisioning
 * progresses.
 */
export interface OrganizationStorage {
    /**
     * The Azure subscription holding the account, or null when it is not hosted in Azure.
     */
    subscriptionId: string | null
    /**
     * The name of the account.
     */
    accountName: string | null
    /**
     * The blob service endpoint of the account, or null until it exists.
     */
    blobEndpoint: string | null
    /**
     * The one host a publish workload may reach, where the account answers the platform: its
     * private endpoint's address inside the platform network, or its public host where no
     * private endpoint exists. Null until the account exists.
     */
    publishHost: string | null
    /**
     * Where the storage is in its lifecycle.
     */
    status: DeploymentStatus
}

File: kinotic-domain/src/main/java/org/kinotic/domain/api/model/OrganizationStorage.java
package org.kinotic.domain.api.model;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.Accessors;

/**
 * An organization's storage: the one account everything its deployments publish goes to,
 * and where that account is in its lifecycle. Held by {@link Organization} once its
 * provisioning job has started; the account's details fill in as provisioning progresses.
 */
@Getter
@Setter
@Accessors(chain = true)
@NoArgsConstructor
public class OrganizationStorage {

    /**
     * The Azure subscription holding the account, or {@code null} when it is not hosted in
     * Azure.
     */
    private String subscriptionId;

    /**
     * The name of the account.
     */
    private String accountName;

    /**
     * The blob service endpoint of the account, or {@code null} until it exists.
     */
    private String blobEndpoint;

    /**
     * The one host a publish workload may reach, where the account answers the platform: its
     * private endpoint's address inside the platform network, or its public host where no
     * private endpoint exists. {@code null} until the account exists.
     */
    private String publishHost;

    /**
     * Where the storage is in its lifecycle.
     */
    private DeploymentStatus status;

}
